detd/systemconf.py

```diff
@@ -203,6 +203,19 @@ def setup_listener(self, interface, mapping, scheduler, stream, maddress, hints)
         except subprocess.CalledProcessError:
             raise
 
+    def set_interface_up(self, interface, retries=5):
+        sysinfo = SystemInformation()
+        ip = CommandIp()
+
+        for attempt in range(1, retries + 1):
```

Comment:
This would always execute setting the interface up all tries. I guess it would be more elegant to just exit once we detect the link.